### PR TITLE
📊 demography: life tables from UN, combined with HMD

### PR DESCRIPTION
**Pull request by lucasrodes**
**New datasets:**
- Add `un_wpp_lt`: UN WPP period life tables (new garden dataset)
- Add `life_tables`: Combined life tables (UN WPP + HMD) (new garden and grapher dataset)

**Tooling:**
- Add `paths.log` to `etl.helpers`:
  - Calls `structlog` under the hood, and adds the step name to the log message. Example:
  ```
  paths.log.info("some message")
  2023-10-03 13:29:12 [info     ] some message                   step=demography/garden/2023-10-03/life_tables
  ```
  - An advantage is that there is no need to have an extra import in the step script (`from structlog import get_logger`). I may replace these imports in another PR.

### CODE REVIEW

File: snapshots/un/2023-10-02/un_wpp_lt_all.zip.dvc [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    version_producer: 2022
+    title_snapshot: World Population Prospects - Life Tables (Both sexes)
+    description_snapshot: |-
+      Provides single-age life tables up to age 100 for both sexes with a set of values showing the mortality experience of a hypothetical group of infants born at the same time and subject throughout their lifetime to the specific mortality rates of a given year. The following series are provided: age-specific mortality rates (mx), probabilities of dying (qx), probabilities of surviving (px), number surviving (lx), number dying (dx), number of person-years lived (Lx), survivorship ratios (Sx), cumulative stationary population (Tx), average remaining life expectancy (ex) and average number of years lived (ax).
+
+    # Citation
+    producer: United Nations
+    citation_full: |-
+      United Nations, Department of Economic and Social Affairs, Population Division (2022). World Population Prospects 2022, Online Edition.
+    attribution: United Nations - World Population Prospects 2022 (2022)
```
Comment (spoonerf) [inline snapshots/un/2023-10-02/un_wpp_lt_all.zip.dvc:23]: Maybe the double 2022 is not needed?

File: etl/steps/data/garden/demography/2023-10-03/life_tables.meta.yml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+    For period data, data prior to 1950 is from HMD, and data from 1950 onwards is from UN WPP.
+
+
+# Learn more about the available fields:
+# http://docs.owid.io/projects/etl/architecture/metadata/reference/tables/
+tables:
+  life_tables:
+    variables:
+      central_death_rate:
```
Comment (spoonerf) [inline etl/steps/data/garden/demography/2023-10-03/life_tables.meta.yml:20]: I am not able to run this locally as I think the metadata is missing for `probability_of_survival` and `survivorship_ratio`. For me, they have the same titles in VariableMeta as a couple of the other variables. They are the last two in this screenshot: 
Screenshot 2023-10-03 at 14 29 17

Comment (spoonerf) [reply etl/steps/data/garden/demography/2023-10-03/life_tables.meta.yml:20]: Or somehow the metadata isn't propagating as expected, not sure.

Comment (spoonerf) [reply etl/steps/data/garden/demography/2023-10-03/life_tables.meta.yml:20]: Might just be local to me as the CI is passing - have you been able to get the data onto staging @lucasrodes?

File: etl/steps/data/garden/un/2023-10-02/un_wpp_lt.meta.yml [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        title: Number of deaths
+        short_unit: deaths
+        unit: deaths
+        description_short: Number of deaths between ages x and x+n.
+      number_person_years_lived:
+        title: Number of person-years lived
+        unit: person-years
+        description_short: Survival ratio corresponding to proportion of the life table population in age group (x, x+n) who are alive n year later.
+      survivorship_ratio:
+        title: Number of person-years lived
```
Comment (spoonerf) [inline etl/steps/data/garden/un/2023-10-02/un_wpp_lt.meta.yml:58]: Ah @lucasrodes  - the issue is here, this variable title is duplicated and `probability_of_survival` is too

File: snapshots/un/2023-10-02/un_wpp_lt_all.zip.dvc (hunk 1 above)
Comment (lucasrodes) [reply snapshots/un/2023-10-02/un_wpp_lt_all.zip.dvc:23]: Yeah, not great.

I copied it from their suggested citation ([see the footer](https://population.un.org/wpp/Download/Standard/MostUsed/)), so I'd leave it as is for now.

File: etl/steps/data/garden/un/2023-10-02/un_wpp_lt.meta.yml (hunk 3 above)
Comment (lucasrodes) [reply etl/steps/data/garden/un/2023-10-02/un_wpp_lt.meta.yml:58]: oh wow, nice catch, Fi!

Comment (spoonerf) [reply etl/steps/data/garden/un/2023-10-02/un_wpp_lt.meta.yml]: Maybe this is something that should fail on the CI? Or would that be a big pain to implement?

Comment (Marigold) [reply etl/steps/data/garden/un/2023-10-02/un_wpp_lt.meta.yml]: I think we have a dataset that uses different variables with same titles. (We could raise a warning though)

**Pull request merged**